app/controllers/api/v1/orders_controller.rb

```diff
@@ -14,6 +14,20 @@ def show
                  status: 404
         end
       end
+
+      def index
+        order_references = Order::Reference.where(search_query)
+
+        render :index, locals: { order_references: order_references }
+      end
+
+      private
+      
+      def search_query
+        OrderQuery.new(warehouse_id: params[:warehouse_id],
```

Comment:
Oh the other concern I have is anytime you query with just the OrderNumber it's super slow, so I want to prevent that. But Ransack may have features to deal with that.